modules/database/src/std/rec/mbboDirectRecord.c

```diff
@@ -234,6 +219,9 @@ static long process(struct dbCommon *pcommon)
         recGblGetTimeStampSimm(prec, prec->simm, NULL);
     }
 
+    /* update bits to reflect any change made by dset */
+    bitsFromVAL(prec);
```

Comment:
Noting behavior change: Earlier versions didn't set the Bn fields when VAL was read from DOL (with OMSL=closed_loop).

Reply:
I think that's a desirable behavior change. Setting Bn from VAL whenever VAL is set from DOL (closed_loop) or via channel access (supervisory) seems useful. Side effect would be additional monitors, but only if there are network clients, and they are very likely there because some display shows Bx, so they're expected to update.

Reply:
+1

Reply:
I agree that this change is desirable. It should be documented though.